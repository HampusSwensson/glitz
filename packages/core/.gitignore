node_modules/
es/
server/