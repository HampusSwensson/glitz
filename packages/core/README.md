# ![Glitz](https://github.com/frenic/glitz/raw/master/glitz.svg?sanitize=true)

The core implementation for Glitz. Head over to [https://github.com/frenic/glitz](https://github.com/frenic/glitz) for details.
