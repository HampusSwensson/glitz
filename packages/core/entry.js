// Entry file for bundle

export { default as GlitzClient } from './src/client/GlitzClient.ts';
