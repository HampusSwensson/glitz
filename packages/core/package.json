{
  "name": "@glitz/core",
  "version": "1.1.5",
  "main": "./cjs/index.js",
  "module": "./jsm/index.js",
  "esnext": "./esnext/index.js",
  "types": "./esnext/index.d.ts",
  "license": "MIT",
  "author": "Fredrik Nicol <fredrik.nicol@gmail.com>",
  "homepage": "https://github.com/frenic/glitz",
  "repository": "git@github.com:frenic/glitz.git",
  "bugs": "https://github.com/frenic/glitz/issues",
  "side-effects": false,
  "dependencies": {
    "@glitz/type": "^1.1.4"
  },
  "scripts": {
    "clean": "rimraf cjs esnext jsm",
    "build": "tsc -p tsconfig.build.json && node ../../bundle.js src/index.ts --cjsx2 index --jsm index && node ../../bundle.js src/server.ts --cjs server --jsm server",
    "prepublish": "npm run clean & npm run build"
  },
  "files": [
    "cjs/",
    "esnext/",
    "jsm/",
    "server/"
  ]
}
