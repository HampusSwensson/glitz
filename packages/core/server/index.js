module.exports = require('../cjs/glitz.server');
