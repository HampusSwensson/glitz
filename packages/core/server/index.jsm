export * from '../jsm/server';
export { default } from '../jsm/server';
