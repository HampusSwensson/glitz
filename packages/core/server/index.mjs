export * from '../mjs/server';
export { default } from '../mjs/server';
