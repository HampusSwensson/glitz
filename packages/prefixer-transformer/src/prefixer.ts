// @ts-ignore
import { Style } from '@glitz/core';
import * as prefixer from 'inline-style-prefixer/static';

export default prefixer as typeof prefixer;
