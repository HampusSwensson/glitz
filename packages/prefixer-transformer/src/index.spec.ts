import { GlitzServer } from '@glitz/core';
import prefixer from './';

describe('prefixer', () => {
  it('injects prefixed atomic style', () => {
    const server = new GlitzServer({ transformer: prefixer });

    expect(server.injectStyle({ display: 'flex' })).toBe('a');
    expect(server.getStyle()).toMatchInlineSnapshot(
      `".a{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}"`,
    );
  });
});
