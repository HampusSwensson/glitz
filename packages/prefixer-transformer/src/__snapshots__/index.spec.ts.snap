// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prefixer injects prefixed atomic style 1`] = `"<style data-glitz>.a{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}</style>"`;

exports[`prefixer injects prefixed non-atomic style 1`] = `"<style data-glitz>.a{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}</style>"`;
